tests/math/test_trace_distance_math.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for differentiable trace distance.
+"""
+
+import numpy as onp
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+pytestmark = pytest.mark.all_interfaces
+
+tf = pytest.importorskip("tensorflow", minversion="2.1")
+torch = pytest.importorskip("torch")
+jax = pytest.importorskip("jax")
+jnp = pytest.importorskip("jax.numpy")
+
+
+class TestTraceDistanceMath:
+    """Tests for the trace distance function between two states (state vectors or density matrix)."""
+
+    state0_state1_td = [
+        # Mat-Mat-TD
+        (([[1, 0], [0, 0]], [[0.5, 0], [0, 0.5]]), 0.5),
+        (([[0, 0], [0, 1]], [[0.5, 0], [0, 0.5]]), 0.5),
```

Comment:
Nitpicking a little bit, but can we add more variety in the expected values? At the very least we should check that same states give 0 trace distance and orthogonal states give 1.